lib/renderer/window-setup.ts

```diff
@@ -0,0 +1,281 @@
+          const guestURL = getGuestURL()
+          if (guestURL) {
+            // TypeScript doesn't want us to assign to read-only variables.
+            // It's right, that's bad, but we're doing it anway.
+            (guestURL as any)[property] = newVal
+
+            return ipcRenderer.sendSync(
+              'ELECTRON_GUEST_WINDOW_MANAGER_WEB_CONTENTS_METHOD_SYNC',
+              guestId, 'loadURL', guestURL.toString())
+          }
+        }
+      }
+    }
+
+    defineProperties(this, {
+      hash: propertyProxyFor<string>(LocationProperties.hash),
+      href: propertyProxyFor<string>(LocationProperties.href),
+      host: propertyProxyFor<string>(LocationProperties.host),
+      hostname: propertyProxyFor<string>(LocationProperties.hostname),
+      origin: propertyProxyFor<string>(LocationProperties.origin),
+      pathname: propertyProxyFor<string>(LocationProperties.pathname),
+      port: propertyProxyFor<string>(LocationProperties.port),
+      protocol: propertyProxyFor<string>(LocationProperties.protocol),
+      search: propertyProxyFor<URLSearchParams>(LocationProperties.search)
+    })
+  }
+
+  public toString (): string {
+    return this.href
+  }
+}
+
+class BrowserWindowProxy {
+  public closed: boolean = false
+
+  private _location: LocationProxy
+
+  // TypeScript doesn't allow getters/accessors with different types,
+  // so for now, we'll have to make do with an "any" in the mix.
+  // https://github.com/Microsoft/TypeScript/issues/2521
+  public get location (): LocationProxy | any {
+    return this._location
+  }
+  public set location (url: string | any) {
+    url = resolveURL(url)
+    this.ipcRenderer.sendSync('ELECTRON_GUEST_WINDOW_MANAGER_WEB_CONTENTS_METHOD_SYNC', this.guestId, 'loadURL', url)
+  }
+
+  constructor (private ipcRenderer: Electron.IpcRenderer, private guestId: any) {
+    this._location = new LocationProxy(ipcRenderer, guestId)
+
+    ipcRenderer.once(`ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_CLOSED_${guestId}`, () => {
+      removeProxy(guestId)
+      this.closed = true
+    })
+  }
+
+  public close () {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_CLOSE', this.guestId)
+  }
+
+  public focus () {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_METHOD', this.guestId, 'focus')
+  }
+
+  public blur () {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_METHOD', this.guestId, 'blur')
+  }
+
+  public print () {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WEB_CONTENTS_METHOD', this.guestId, 'print')
+  }
+
+  public postMessage (message: any, targetOrigin: any) {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_POSTMESSAGE', this.guestId, message, toString(targetOrigin), window.location.origin)
+  }
+
+  public eval (...args: any[]) {
+    this.ipcRenderer.send('ELECTRON_GUEST_WINDOW_MANAGER_WEB_CONTENTS_METHOD', this.guestId, 'executeJavaScript', ...args)
+  }
+}
+
+const windowSetup = (
```

Comment:
Can we `export const` instead of `export default` further down the file.  It's an IMO thing but I much prefer enforcing naming conventions and when we still have `require` the `default` export is gonna look really weird

Reply:
I agree with that, I'll change it.